03_python_fastapi_project/main.py

```
@@ -113,7 +128,91 @@ async def delete_product(product_id: int, db: AsyncSession = Depends(get_db)):
     await db.commit()
     return {"message": "Product deleted successfully"}
 
+@app.get("/cart/items", response_model=List[CartItemResponse])
+async def get_cart_items(db: AsyncSession = Depends(get_db)):
+    result = await db.execute(select(CartItem).options(joinedload(CartItem.product)))
+    cart_items = result.scalars().all()
+    return cart_items
+
+@app.post("/cart/items", response_model=CartItemResponse)
+async def add_to_cart(cart_item: CartItemDTO, db: AsyncSession = Depends(get_db)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)